src/utils/postErrorToSlack.js

```diff
@@ -0,0 +1,98 @@
+import Auth from '@aws-amplify/auth';
+import { getLoggerBotToken } from 'utils/crypt';
+
+const extractExperimentId = (url) => {
+  const match = url.match(/experiments\/([^/]+)/i);
+  return match ? match[1] : null;
+};
+
+const takeNumLines = (string, numLines) => string.split('\n').slice(0, numLines).join('\n');
+
+const buildErrorMessage = (error, componentStack, reduxDump, context) => {
+  const {
+    user, timestamp, experimentId, url,
+  } = context;
+
+  return (`
+    Uncaught UI Error - Exp ID ${experimentId} - ${timestamp}
+
+    === DETAILS ===
+    Name: ${user.attributes.name}
+    Email: ${user.attributes.email}
+    UserID: ${user.username}
+    ExperimentID: ${experimentId}
+    URL: ${url}
+    Timestamp: ${timestamp}
+
+    ===== ERROR =====
+    ${error.toString()}
+    ${takeNumLines(componentStack, 11)}
+
+    ===== REDUX STATE =====
+    ${JSON.stringify(reduxDump, null, 2)}`
+  );
+};
+
+const postError = async (errorLog, context) => {
+  const {
+    user, timestamp, experimentId, url,
+  } = context;
+
+  const message = `
+  \u26A0  Uncaught UI Error - ExpID ${experimentId} - ${timestamp}
+  URL: ${url}
+
+  User: ${user.attributes.name}
+  Email: ${user.attributes.email}
+  User ID: ${user.username}
```

Comment:
Can you not get the message from `buildErrorMessage` and just pass it here without creating a new message variable with those fields again? I see there are repeating  values in both places

Reply:
`buildErrorMessage` builds the string that is going to be posted as the error log, which appears in the collapsible part of the error. These lines build the text that appears in the post message - it's primary purpose is to help distinguish one error message from another. So they're for different purposes, although they might look the same.

Reply:
I would probably collapse the user info into a single line
`User: ${user.attributes.name} (${user.attributes.email}) ${user.username}`